Algorithm.Framework/Risk/MaximumDrawdownPercentPortfolio.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.Framework.Risk
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IRiskManagementModel"/> that limits the drawdown
+    /// of the portfolio to the specified percentage - it terminates the algorithm if this limt is reached
+    /// </summary>
+    public class MaximumDrawdownPercentPortfolio : RiskManagementModel
+    {
+        private readonly decimal _maximumDrawdownPercent;
+        private decimal _startingValue;
+        private bool _initialised = false;
+        private string _email;
+        private string _phoneNumber;
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MaximumDrawdownPercentPortfolio"/> class
+        /// </summary>
+        /// <param name="maximumDrawdownPercent">The maximum percentage drawdown allowed for algorithm portfolio
+        /// compared with starting value, defaults to 5% drawdown</param>
+        /// <param name="email">Optional - Email address for Algorithm Termination Notifications</param>
+        /// <param name="phoneNumber">Optional - Phone Number for Algorithm Termination SMS Notifications</param>
+        public MaximumDrawdownPercentPortfolio(
+            decimal maximumDrawdownPercent = 0.05m, string email = "", string phoneNumber = ""
+            )
+        {
+            _maximumDrawdownPercent = -Math.Abs(maximumDrawdownPercent);
+            _email = email;
+            _phoneNumber = phoneNumber;
```

Comment:
Despite the fact that it is optional, the pattern that we are following does not include notifications. An user can create a new class that inherits from this one and add the notification helpers if he wants. Or he can look for orders event in `OnOrdersEvent` and send the notification from there.

Reply:
For me the ideal implementation would be for the user to specify their contact details (Email, SMS or both) for "Critical Messages" or maybe just "Errors" and then in the Framework model the developer can output message to this logging level.

Reply:
We have been trying to separate concerns with the framework models. The risk model, for instance, will just create `PortfolioTarget` objects that reduce the Quantity of each asset.